### PR TITLE
Notifications tab with friend request

### PR DESCRIPTION
**Pull request by hayleyowen**
Adds a notifications tab with a friend request notification where you can accept or reject friend requests

### CODE REVIEW

**Review by seanb0 — APPROVED** [review]
Looks Good, I dont think it should cause any problems.  Was it tested before hand?

File: 360/watchapp/templates/watchapp/homepage.html [hunk 1]
```diff
@@ -955,20 +994,6 @@ <h2>{{ friend.username }}</h2>
         setInterval(updateDarkMode, 100);
     </script>
     <script> 
-        // Toggle the dropdown menu visibility
-        function showSettings() {
```
Comment (seanb0) [inline 360/watchapp/templates/watchapp/homepage.html:933]: Do we not need this function anymore?

File: 360/watchapp/templates/watchapp/homepage.html [hunk 2]
```diff
@@ -1013,28 +1038,6 @@ <h2>{{ friend.username }}</h2>
             localStorage.setItem('isOnline', isOnline.checked.toString());
         });
     </script>
-    
-    <script>
-        // script to generate notifications dropdown menu generated from copilot
-
-        $(document).ready(function() {
-            $('#notifications-button').click(function() {
```
Comment (seanb0) [inline 360/watchapp/templates/watchapp/homepage.html]: Or any of this?

File: 360/watchapp/views.py [hunk 3]
```diff
@@ -208,13 +208,7 @@ def homepage_view(request):
     pending_requests_data = [{'request_id': req.id, 'id': req.from_user.id, 'username': req.from_user.username} for req in pending_requests]
     
     # Create a list of usernames from the friends queryset
-    friends_data = [{'username': friend.username} for friend in friends]
-
-    # Retrieve only the videos for the currently logged-in user
-    user_videos = YouTubeData.objects.filter(user=request.user).order_by('-added_at')
```
Comment (seanb0) [inline 360/watchapp/views.py]: Dont need this?

File: 360/watchapp/views.py [hunk 4]
```diff
@@ -230,19 +224,15 @@ def homepage_view(request):
             update_session_auth_hash(request, user)  # Keep user logged in after password change
 
         user.save()
-        return redirect('login')
-    
-    return render(request, 'watchapp/homepage.html', {
```
Comment (seanb0) [inline 360/watchapp/views.py]: Dont need this?

File: 360/watchapp/views.py [hunk 5]
```diff
@@ -230,19 +224,15 @@ def homepage_view(request):
...
-        "friends": friends_data,
-        "pending_requests": pending_requests_data,
-        'user_videos': user_videos
-    })
+        return redirect('login')  # Redirect to login page after changes
 
-def logout_view(request):
```
Comment (seanb0) [inline 360/watchapp/views.py]: Dont need this?

**Review by seanb0 — COMMENTED** [review]
A lot of these were deleted, and nothing was replaced. Just want to make sure it wasnt needed.

File: 360/watchapp/templates/watchapp/homepage.html (hunk 1 above)
Comment (hayleyowen) [reply 360/watchapp/templates/watchapp/homepage.html:933]: no we don't

File: 360/watchapp/templates/watchapp/homepage.html (hunk 2 above)
Comment (hayleyowen) [reply 360/watchapp/templates/watchapp/homepage.html]: moved it up and added the settings dropdown with it

File: 360/watchapp/views.py (hunk 4 above)
Comment (hayleyowen) [reply 360/watchapp/views.py]: it was a duplicate

File: 360/watchapp/views.py (hunk 5 above)
Comment (hayleyowen) [reply 360/watchapp/views.py]: it was a duplicate

**Comment by seanb0** [comment]
All good.

**Review by seanb0 — APPROVED** [review]
All good

**Pull request merged**